### ISSUE TITLE
Font size is not applied in morphology panel - iphone xr

### ISSUE THREAD

**Issue by monzug** [alpheios-project/components#766]
while testing #522 and changing small, medium large font in Options, I noticed that the morphology (or lemma) panel does not change size when font is set to small or large.

**Comment by monzug** [comment]
Looks like the font is applied for the first lookup but then it goes back to medium font. see screenshots below

![Image-1](https://user-images.githubusercontent.com/41396793/63440665-3df3ad00-c3fe-11e9-8479-c92b2dd5807c.jpg)

![Image-1 (1)](https://user-images.githubusercontent.com/41396793/63440672-4350f780-c3fe-11e9-8a81-980a2edf5ae6.jpg)

**Comment by irina060981** [comment]
I checked - it is easily reproducable even in Chrome emulator.

There are two problems here:

1. We have some inconsistence in calc font-size, sometimes we use `--alpheios-base-text-size`, sometmes - `--alpheios-base-ui-size`. But we change only this variable - `--alpheios-base-text-size`

2. When we check any variant of font-size in options - it adds `style="--alpheios-base-text-size:;"`
but when we close the panel - it removes it - and font-size becomes medium.

**Comment by balmas** [comment]
the two size variables are so we can scale ui elements (buttons, etc. ) separately from the text.  We should (I think) always be using --alpheios-base-text-size for any text. We don't currently offer a way for the user to change --alpheios-base-ui-size. 

I cannot think why we might be resetting the scaled font size upon closing the panel though.

**Comment by balmas** [comment]
and I see I used the wrong variable for defining the size of the short definition text....
(@monzug that's why the text of the short definition isn't scaling with the rest of the text in the popup/panel)

**Comment by monzug** [comment]
glad you found a fix.

**Comment by kirlat** [comment]
>     When we check any variant of font-size in options - it adds `style="--alpheios-base-text-size:;"`
>      but when we close the panel - it removes it - and font-size becomes medium.

This code was added to work together with the two functions below it: https://github.com/alpheios-project/components/blob/master/src/vue/components/panel-compact.vue#L570-L581. It is for a pinch zoom support. Before, we were using a font-size CSS var for scaling caused by the pinch zoom action and it was safe to erase all the zoom changes once the panel is closed. We did not want to remember custom zooming between the open panel sessions.

Now, we're using a font-size var for applying the size value from the options too and the code needs to be updated. I think we should change it to the one that will reset the font size value to the one specified in the options (i.e. small/medium/large). Also, it will be good to add a prop that will track if pinch zoom did take place and reset the font size in that case only to make it more efficient.

I can do that change if nobody object.

**Comment by irina060981** [comment]
> Now, we're using a font-size var for applying the size value from the options too and the code needs to be updated. I think we should change it to the one that will reset the font size value to the one specified in the options (i.e. small/medium/large). Also, it will be good to add a prop that will track if pinch zoom did take place and reset the font size in that case only to make it more efficient.

I think it is a good suggestion!

**Comment by kirlat** [comment]
> yes @kirlat please go ahead with that change. Thank you!

Just submitted a fix for that

**Comment by monzug** [comment]
I can't get to the Options panel on iPhone xr in latest build, so I cannot test this issue. maybe a side effect f this change? see screenshot 
![IMG_3940](https://user-images.githubusercontent.com/41396793/63958189-e4236080-ca57-11e9-96f8-6ecca9eaf990.PNG)

**Comment by irina060981** [comment]
Merged and closed

**Comment by balmas** [comment]
components 1.2.35

**Comment by monzug** [comment]
I can still see the empty toolbar with the close button on iPhone xr. see screenshot
![IMG_3955](https://user-images.githubusercontent.com/41396793/64117655-ba25b300-cd63-11e9-822b-1156f102ba91.PNG)

**Comment by monzug** [comment]
also on Samsung 5s
![IMG_3956](https://user-images.githubusercontent.com/41396793/64125985-fe23b280-cd79-11e9-8341-b42695b215c5.jpg)

**Comment by irina060981** [comment]
@balmas , I see the same (that's why I couldn't work on mobile video)
But I couldn't fix it - because for now alpheios-nemo-ui repo is failing on startup for me:

![image](https://user-images.githubusercontent.com/12377640/64169618-88126f00-ce91-11e9-8775-fbbdf8204764.png)

It is because of not enough environment variables.

Bridget, could you help me to fix it? I have no ideas how to solve it.

**Comment by balmas** [comment]
@irina060981 try putting a file named `env.js` in your alpheios_nemo_ui/data/assets directory, with the following contents:

```
onst serverEnv = {
    LOGIN_URL: 'http://dev.alpheios.net:5000/login?next={FROM_URL}',
    LOGOUT_URL: 'http://dev.alpheios.net:5000/logout?next={FROM_URL}',
    SESSION_URL: 'http://dev.alpheios.net:5000/userinfo',
    TOKEN_URL: 'http://dev.alpheios.net:5000/usertoken',
    ENDPOINTS: {
      wordlist : 'https://userapis.alpheios.net/v1/words'
    }
}
```
~

**Comment by balmas** [comment]
correction: in the alpheios_nemo_ui/data/assets/js directory

Although I am not sure that is the source of the problem because I get a different error if that file is missing.

**Comment by balmas** [comment]
@irina060981 the alpheios code should not throw errors if the auth environment is not defined, and I have seen that  to be true but I'm thinking maybe that something is going wrong with the registration of the AuthModule. 

embed-lib has this code:

```
  if (this.authEnv) {
      if (authEnv.CLIENT_ID) {
        // Register an authentication module only with authentication environment is loaded
        this.ui.registerModule(components.AuthModule, { auth: new AppAuthenticator(authEnv) })
      } else if (authEnv.LOGIN_URL) {
        this.ui.registerModule(components.AuthModule, { auth: new SessionAuthenticator(authEnv) })
      }
    } else {
      this.ui.registerModule(components.AuthModule, { auth: null })
    }
```
so I would guess this is where something is going wrong in your environment.

**Comment by irina060981** [comment]
Oh, thank you, Bridget!
I have guessed what was the problem!

There are two `env.js` files - in the root and in the `data/assets`
I was updating data in the root file (according to the `setup_env.bat` - it doesn't work automatically, but I copied manually).
And I didn't guess that there is another one in data/assets

After updating there  - it started to work.
Thank you :)

**Comment by irina060981** [comment]
@balmas, I have checked the texts-test.alpehios.net on 
   IPhone
   Samsung
   
in both browsers

And all of them shows the same bug as @monzug  reported.
But locally (I have updated alpheios-nemo-ui with `npm i` and `npm up`) it works well and shows correctly

![image](https://user-images.githubusercontent.com/12377640/64175975-d0388e00-ce9f-11e9-9fac-6ee7f44ecfd6.png)

May be there are some problems with update on the server?
@balmas , do you see the correct version on your phone?

**Comment by balmas** [comment]
hmm, yes maybe I messed it up when I rebuilt components. Will look into it.

**Comment by balmas** [comment]
ah, it's not components. I broke it with the simple mode changes in embed-lib 3.0.0.37.  I will fix it.

**Comment by balmas** [comment]
ok. should be fixed now with embed-lib 3.0.0.38

**Comment by monzug** [comment]
tested on iPhone xr, 3.0.0.38. verified.
